aggregator/aggregator.go

```diff
@@ -404,6 +444,29 @@ func (agg *aggregator) ownedShards() []*aggregatorShard {
 	return ownedShards
 }
 
+func (agg *aggregator) placement() (services.Placement, error) {
+	stagedPlacement, onStagedPlacementDoneFn, err := agg.placementWatcher.ActiveStagedPlacement()
+	if err != nil {
+		return nil, err
+	}
+	placement, onPlacementDoneFn, err := stagedPlacement.ActivePlacement()
```

Comment:
Perhaps instead of explicitly calling, after dealing with `return nil, err` you can just above calling `ActivePlacement()` perform a `defer onStagedPlacementDoneFn()`